### ISSUE TITLE
Error "Znuny detected possible network issues." when opening an activity dialog

### ISSUE THREAD

**Issue by schnudd31do3** [znuny/Znuny#228]
### Environment

- OS: Debian Bullseye
- Browser: Chrome
- Znuny version: 6.3.1

### Expected behaviour

No error message when opening an activity dialog

### Actual behaviour

When opening a activity dialog, the Error "Znuny detected possible network issues." appears.
![image](https://user-images.githubusercontent.com/72391822/159253671-88fb7526-6a61-479f-8d76-686ce2df3a98.png)

After a couple of seconds, the message changes to:
![image](https://user-images.githubusercontent.com/72391822/159274608-58c63d3a-60a9-4290-8e1f-07266c7c6e4a.png)

### Additional information

I compared the version 6.2.2 to 6.3.1 and found what makes the difference:
File: /opt/otrs/Kernel/Output/HTML/Templates/Standard/ProcessManagement/ActivityDialogHeader.tt

Version 6.2.2, line 62:
`            <input type="hidden" name="ProcessEntityID" value="[% Data.ProcessEntityID | html %]"/>
`

Version 6.3.1, line 62:
`            <input type="hidden" ID="ProcessEntityID" name="ProcessEntityID" value="[% Data.ProcessEntityID | html %]"/>
`

When I use the line 62 from version 6.2.2 in the template of version 6.3.1, the error message has gone.

**Comment by hanneshal** [comment]
Hi @schnudd31do3 
does this also happen on "fresh" Activity dialogs or only on existing ones?

Regards

Edit: The error is most likely shown in the apache error log and / or in the Znuny log. It would help if you can pass us those too.

**Comment by schnudd31do3** [comment]
I have this error with existing and new activity dialogs und no error messages are found in the apache2 log nor in any other logs in /opt/otrs/var/log

**Comment by hanneshal** [comment]
> I have this error with existing and new activity dialogs und no error messages are found in the apache2 log nor in any other logs in /opt/otrs/var/log

thats strange.

Just to be sure. We are talking about processes in the regular Agent GUI, not editing Activity Dialogs in the Admin Interface, correct?

If Znuny throws an error message like this an AJAX request came back with an error code. 
Would you please have a look at the Browser Console and check for a failing request. 

Just a side question: Did you restarted the Webserver after the update?

**Comment by hanneshal** [comment]
Never mind: I could reproduce it

**Comment by schnudd31do3** [comment]
"Just to be sure. We are talking about processes in the regular Agent GUI, not editing Activity Dialogs in the Admin Interface, correct?" -> Yes

"If Znuny throws an error message like this an AJAX request came back with an error code." and "Would you please have a look at the Browser Console and check for a failing request." -> I could not find any error messages anywhere on the system which could have something to do with it.

"Just a side question: Did you restarted the Webserver after the update?" -> Yes I did

**Comment by Kaffee81** [comment]
We also have this in our Testsystem with a znuny 6.3.1 version.
No Errors in Chrome Console
No Errors in Apache Errorlog

There are no chrome network tab entries with bad index.pl querys. Everything is code 200 and and show a good response (either json or html).

The error message immediately corrects itself and shows a green check icon instead of the yellow powercord icon.

**Comment by hanneshal** [comment]
The issue is related to a JS AJAX Call which should only be used at the beginning of a process. The call is triggered by an onChange Event which is set to the ID Element of "ProcessEntityID".

A fixed file, can be found here: 
[ActivityDialogHeader.tt.zip](https://github.com/znuny/Znuny/files/8317178/ActivityDialogHeader.tt.zip)

As @schnudd31do3 suggested the ID field is the trigger for this AJAX request and results in the behaviour. 

We are preparing a release in the next days to fix this and other issues. If you stumble upon this, you can fix it following this steps:
1) Create this folder: 
```
mkdir -p Custom/Kernel/Output/HTML/Templates/Standard/ProcessManagement/
```
2)As OTRS User: Copy and unzip the file attached to this comment into the new folder. 
3) Restart the webserver

OR

1) Create this folder: 
```
mkdir -p Custom/Kernel/Output/HTML/Templates/Standard/ProcessManagement/
```
2) Copy file from the framework to this folder:
```
cp Kernel/Output/HTML/Templates/Standard/ProcessManagement/ActivityDialogHeader.tt Custom/Kernel/Output/HTML/Templates/Standard/ProcessManagement
```

3) Modify Line 62:
FROM: 
`            <input type="hidden" ID="ProcessEntityID" name="ProcessEntityID" value="[% Data.ProcessEntityID | html %]"/>
`
TO:
`            <input type="hidden" name="ProcessEntityID" value="[% Data.ProcessEntityID | html %]"/>`

Thanks to @schnudd31do3 for reporting this.

Sorry for the inconvenience.

**Comment by schnudd31do3** [comment]
"We are preparing a release in the next days to fix this and other issues." -> I found other issues in the nightly build znuny-6.3.1.22.79. Therefore I'm looking forward to the new version 👍

Edit:
Do you plan also to implement my feature request: https://github.com/znuny/znuny-feature-requests/issues/6 for the described release?

**Comment by hanneshal** [comment]
Hi @schnudd31do3 

if you find new issues, not reported here, it would help us if you create issues for them or just send us a list bugreport@znuny.com if you don't want to go too much in detail here. 
Thanks, your help is appreciated. 

Johannes
Note: 
please don't mix issues and non related feature requests. If a FR is accepted it gets the "Work in progress" Tag. If a release is known it gets a release milestone.

**Comment by schnudd31do3** [comment]
Hi @hanneshal,
thanks for your practical hints to solve. I will discuss my new observations in the forum.

**Comment by paulfolkers** [comment]
Hi everyone, 

we just discovered this problem in a customer project.
We solved the issue another way, from our pov this seems to be a "cleaner" solution.
Since we expect the "ID" part in the ActivityDialogHeader.tt to be useful for something (otherwise that would not have been committed in 6.3 development), we decided to implement a check in the var/httpd/htdocs/js/Core.Agent.TicketProcess.js on line 189 and changed it from:
`if ($('#ProcessEntityID').val() !== "") {`
to
`if ($('#ProcessEntityID').val() !== "" && typeof $('input[name=ActivityDialogEntityID]').val() === 'undefined') {`.

That prevents triggering the "change" function in case we are working in an ActivityDialog for an already enrolled process. If the change function is not being executed, you do not get the error. 

@hanneshal If you have any further questions concerning this, you can contact us as **Sector Nord AG** via the known ways (e.g. Daniel) or just answer this comment.

**Comment by hanneshal** [comment]
Hi @paulfolkers 

> otherwise that would not have been committed in 6.3 development
this was a relict from an older draft of process scopes. It is not needed anymore. 
The first draft was a bit more complex and @dennykorsukewitz shredded it all down to the needed minimum and switched everything down to just "YAM Config" 

Thanks for your solution. I pass this on to @dennykorsukewitz & @jepf  and the they will decide which one they take.

Regards

**Comment by paulfolkers** [comment]
> this was a relict from an older draft of process scopes. It is not needed anymore.
> The first draft was a bit more complex and @dennykorsukewitz shredded it all down to the needed minimum and switched everything down to just "YAM Config"

Good to know, I think both bugfixes are pretty "clean" then. Thanks for letting us know. 

> Thanks for your solution. I pass this on to @dennykorsukewitz & @jepf and the they will decide which one they take.

Alright, thank you!

**Comment by hanneshal** [comment]
Fixed in 6.3.2+
